### PR TITLE
PWA-2375 Improved version banner

### PR DESCRIPTION
**Pull request by JustinConabreeAbsolunet**
## Description

Adding information about the installed PWA Studio packages to the build output. This will make it easier to determine what versions production sites are running for debugging and statistical purposes.

## Related Issue

Closes PWA-2375

### Verification Stakeholders

@supernova-at 

#### Test scenario(s) for direct fix/feature

1.  Run build (or skip to step 2 if verifying on PR instance)
2. Verify comment header in dist JS files (ex the client or vendors JS files)
3. Should contain information about the `@magento` and `@apollo` packages that are in use

## Checklist

- [ ] I have added tests to cover my changes, if necessary.
- [ ] ~I have added translations for new strings, if necessary.~
- [ ] ~I have updated the documentation accordingly, if necessary.~

### CODE REVIEW

File: packages/pwa-buildpack/lib/WebpackTools/configureWebpack/getClientConfig.js [hunk 1]
```diff
@@ -199,22 +199,24 @@ async function getClientConfig(opts) {
...
+            ),
+            ...Object.fromEntries(
+                Object.entries(packageJson.devDependencies || {}).filter(
+                    ([packageKey]) => packageRegex.test(packageKey)
+                )
+            )
+        };
+        versionBanner = Object.entries(pwaStudioVersions)
+            .map(([packageKey, version]) => `${packageKey}: ${version}`)
+            .join(', ');
```
Comment (fooman) [inline packages/pwa-buildpack/lib/WebpackTools/configureWebpack/getClientConfig.js:219]:
```suggestion
        versionBanner = projectConfig.section('staging').buildId;
        if (!versionBanner || versionBanner.trim().length === 0) {
            const packageJson = require(path.resolve(context, './package.json'));
            const packageRegex = /^@magento|^@apollo/;
            const pwaStudioVersions = {
                'pwa-studio': packageJson.version,
                ...Object.fromEntries(
                    Object.entries(packageJson.dependencies || {}).filter(
                        ([packageKey]) => packageRegex.test(packageKey)
                    )
                ),
                ...Object.fromEntries(
                    Object.entries(packageJson.devDependencies || {}).filter(
                        ([packageKey]) => packageRegex.test(packageKey)
                    )
                )
            };
            versionBanner = Object.entries(pwaStudioVersions)
                .map(([packageKey, version]) => `${packageKey}: ${version}`)
                .join(', ');
        }    
```

Comment (JustinConabreeAbsolunet) [reply packages/pwa-buildpack/lib/WebpackTools/configureWebpack/getClientConfig.js:219]: CCing in @supernova-at

Comment (supernova-at) [reply packages/pwa-buildpack/lib/WebpackTools/configureWebpack/getClientConfig.js:219]: I'm fine keeping the build ID support in there but I'd prefer it not to be an either / or.

Output all the versions as we have here and if there's a build ID present, we can output that too 👍

I'm thinking about the case where someone accidentally leaves that environment variable set on their production deployment.

File: packages/pwa-buildpack/lib/WebpackTools/configureWebpack/getClientConfig.js [hunk 2]
```diff
@@ -199,22 +199,24 @@ async function getClientConfig(opts) {
...
+                    ([packageKey]) => packageRegex.test(packageKey)
+                )
+            ),
+            ...Object.fromEntries(
+                Object.entries(packageJson.devDependencies || {}).filter(
+                    ([packageKey]) => packageRegex.test(packageKey)
+                )
+            )
+        };
+        versionBanner = Object.entries(pwaStudioVersions)
```
Comment (supernova-at) [inline packages/pwa-buildpack/lib/WebpackTools/configureWebpack/getClientConfig.js:217]: +1 bonus point if we add a `sort()` at the end 😁

Comment (JustinConabreeAbsolunet) [reply packages/pwa-buildpack/lib/WebpackTools/configureWebpack/getClientConfig.js]: Done!
Sample:
```
/*! @version pwa-studio: 12.1.0, build-id: testversion, @magento/venia-ui: ~9.1.0, @magento/upward-security-headers: ~1.0.5, @magento/pwa-theme-venia: ~1.1.0, @magento/pwa-buildpack: ~11.0.0, @magento/peregrine: ~12.1.0, @magento/pagebuilder: ~7.0.1, @magento/eslint-config: ~1.5.0, @magento/babel-preset-peregrine: ~1.2.0, @apollo/client: ~3.4.0 */
```

File: packages/pwa-buildpack/lib/WebpackTools/configureWebpack/getClientConfig.js (hunk 1 above)
Comment (JustinConabreeAbsolunet) [reply packages/pwa-buildpack/lib/WebpackTools/configureWebpack/getClientConfig.js]: I've added the build ID when available!

**Review by michaelyu0123 — APPROVED** [review]
LGTM

**Comment by dpatil-magento** [comment]
QA Approved, waiting for cypress test to assert on client.*.js.

**Review by dpatil-magento — APPROVED** [review]

**Pull request merged**